modules/rostests/kmtests/ntos_ob/ObInsert.c

```diff
@@ -0,0 +1,651 @@
+
+    /* Check the object handles before insertion */
+    ok_eq_ulong(ObjectHeader->PointerCount, 1LU);
+    ok_eq_ulong(ObjectHeader->HandleCount, 0LU);
+
+    /* Insert the object */
+    Status = ObInsertObject(Object,
+                            NULL,
+                            STANDARD_RIGHTS_ALL,
+                            0,
+                            NULL,
+                            &Handle);
+
+    ok_eq_hex(Status, STATUS_OBJECT_PATH_SYNTAX_BAD);
+
+    /* Check the object handles after invalid insertion. ObInsertObject should've
+       automatically dereferenced the object */
+    ok_eq_ulong(ObjectHeader->PointerCount, 0LU);
+    ok_eq_ulong(ObjectHeader->HandleCount, 0LU);
+}
+
+/*!
+ * @brief This function tests STATUS_OBJECT_NAME_COLLISION.
+ */
+static
+VOID
+ObInsert_NameCollision()
+{
+    NTSTATUS Status;
+    UNICODE_STRING ObjectName;
+    PDUMMY_TYPE Object;
+    HANDLE Handle;
+    POBJECT_HEADER ObjectHeader;
+
+    /* Initialize the object name. To simulate STATUS_OBJECT_NAME_COLLISION, provide
+       a name that already exists */
+    RtlInitUnicodeString(&ObjectName,
+                         L"\\");
+
+    /* Initialize object attributes with the name. To simulate STATUS_OBJECT_NAME_COLLISION and
+       avoid STATUS_OBJECT_NAME_EXISTS don't specify OBJ_OPENIF */
+    InitializeObjectAttributes(&DummyObjectAttributes,
+                               &ObjectName,
+                               OBJ_CASE_INSENSITIVE,
+                               NULL,
+                               NULL);
+
+    /* Create the object */
+    Status = ObCreateObject(KernelMode,
+                            DummyType,
+                            &DummyObjectAttributes,
+                            KernelMode,
+                            NULL,
+                            sizeof(DUMMY_TYPE),
+                            0,
+                            0,
+                            (PVOID *)&Object);
+
+    ok_eq_hex(Status, STATUS_SUCCESS);
+
+    /* Initialize the object data */
+    RtlZeroMemory(Object, sizeof(DUMMY_TYPE));
+    Object->Data = 123;
+
+    ObjectHeader = OBJECT_TO_OBJECT_HEADER(Object);
+
+    /* Check the object handles before insertion */
+    ok_eq_ulong(ObjectHeader->PointerCount, 1LU);
+    ok_eq_ulong(ObjectHeader->HandleCount, 0LU);
+
+    /* Insert the object */
+    Status = ObInsertObject(Object,
+                            NULL,
+                            STANDARD_RIGHTS_ALL,
+                            0,
+                            NULL,
+                            &Handle);
+
+    ok_eq_hex(Status, STATUS_OBJECT_NAME_COLLISION);
+
+    /* Check the object handles after invalid insertion. ObInsertObject should've
+       automatically dereferenced the object */
+    ok_eq_ulong(ObjectHeader->PointerCount, 0LU);
+    ok_eq_ulong(ObjectHeader->HandleCount, 0LU);
+}
+
+START_TEST(ObInsert)
+{
+    /*
+     * Initialize a dummy object type
+     */
```

Comment:
```suggestion
    /* Initialize a dummy object type */
```
These are simple mono-line comments, I think you can make them hold on one single line.